### PR TITLE
Feature/display-local-time

### PR DESCRIPTION
**Pull request by ventura-rivera**
This feature adds a timestamp to `wavedata.txt` and `gendata.txt` so that viewers can view the local time to better pinpoint parts of the video they are interested in seeing.

### CODE REVIEW

File: noaagen.py [hunk 1]
```diff
@@ -15,9 +16,10 @@ def print_hi(name):
     file = open('/surf/gendata.txt')
     content = file.readlines()
     data = content[0]+content[1]+content[2]
-    data = re.sub(tlreg, '', data)
-    data = re.sub(mlreg, '', data)
-    data = re.sub(blreg, '', data)
+    timestamp = datetime.now().strftime("%m/%d/%y %H:%M")
+    data = re.sub(tlreg, '  TIMESTAMP', data)
+    data = re.sub(mlreg, '  MM/DD/YY hh:mm TZ', data)
+    data = re.sub(blreg, f'  {timestamp} PST', data)  # datetime.now() returns local time, so assume PST
```
Comment (ventura-rivera) [inline noaagen.py]: Note: I am hardcoding `PST` here with the assumption that the host is running in the PST timezone. There is a potential to use `pytz` but that is out of scope for the Python Standard Library and will require a pip install on the host.

File: noaawave.py [hunk 2]
```diff
@@ -9,7 +10,11 @@ def print_hi(name):
     with open('/surf/wavedata.txt', 'wb') as f:
         f.write(r.content)
     file = open('/surf/wavedata.txt')
+    timestamp = datetime.now().strftime("%m/%d/%y %H:%M")
     content = file.readlines()
+    content[0] = content[0].strip() + "   TIMESTAMP\n"
+    content[1] = content[1].strip() + "  MM/DD/YY hh:mm TZ\n"
+    content[2] = content[2].strip() + f"   {timestamp} PST\n"  # datetime.now() returns local time, so assume PST
```
Comment (ventura-rivera) [inline noaawave.py]: same comment as the one I made in the `noaagen.py` file

**Review by ventura-rivera — COMMENTED** [review]
Just one thing I wanted to note.

**Comment by ventura-rivera** [comment]
@slapplebags I cannot request a reviewer. Can you please check this out?

**Comment by slapplebags** [comment]
I'll give it a go in the morning

**Review by ventura-rivera — COMMENTED** [review]
Updated my logic by creating a separate script to handle the timestamp operations. Updated the `crontab` so that the timestamp can be updated before each time the subtitle file is updated. This way, the timestamp is updated every minute.

I contemplated keeping the timestamp operations in the noaagen.py and noaawave.py scripts, and updating `crontab` to run those scripts before each time the subtitle file is updated, but it seemed excessive to make requests to noaa.gov every minute.

Went down a rabbit hole on this lol. I need to sleep now. Looking forward to your review.

**Comment by slapplebags** [comment]
looks good and I appreciate the work but I had a thought. Any reason we wouldn't use the cameras built in timestamp? I can turn that on if that offers the same functionality you're after.

**Comment by ventura-rivera** [comment]
You know, I thought to myself if this camera has built-in time, and I agree it would be much preferable. I did some digging, and I think I can make that change.

**Comment by ventura-rivera** [comment]
@slapplebags After some digging, I updated the ffmpeg to use the localtime. Apologies if the command looks off at all (I have never used ffmpeg before). If it does, please suggest where it should be altered.

Also, if there is something wrong with the ffmpeg `localtime`, and you want to revert back to the timestamp script, just let me know.

**Comment by slapplebags** [comment]
I've enabled the camera's built in timestamp feature and set it up in the upper right of the image. A lot less overhead for us then adding it to the wave data scraper or the ffmpeg command. Let me know if that is sufficient, if not we can look at merging this in.

**Comment by ventura-rivera** [comment]
oh heck yeah!! It looks great! Thanks for adding that in! I will close this PR.

In the future, if there is anything you'd like some assistance with, I'd be happy to lend a hand 🙂

**Comment by slapplebags** [comment]
Most common request so far is to convert the metric units to imperial, could have a go at that if you're looking for a quick side project.

**Comment by ventura-rivera** [comment]
sounds good, I'll give it a go later on!

**Comment by slapplebags** [comment]
Also if you have any interest i want to build my own wave data buoy a little closer in to shore. I've found an open source project that put some ground work in but theres lots of room for improvement.

**Comment by ventura-rivera** [comment]
Yeah definitely interested! Do you have anything up on Github yet? And where can I find the open source project?

**Comment by slapplebags** [comment]
nothing in github yet but i'll setup a repo and add you. Project i found is here: https://github.com/wattnotions/OpenWave. I've ordered an arduino mkr zero and the same BNO055 IMU he uses, both of which should be here soon. I'm planning on using LoRA for comms.

**Comment by ventura-rivera** [comment]
I don't have a ton of experience with embedded systems and IoT, although I have worked on arduinos and raspberry pis in the past, and it is an area I am _highly_ interested in learning more about. I'll dig though the repo, and learn more about it!

**Comment by slapplebags** [comment]
Not a problem, I've got a fair amount of embedded and iot experience, but
not so much with imu / the required data processing.

**Comment by ventura-rivera** [comment]
Just accepted the invitation to the WaveOSS repo. I'll read more into all of this!
